models/llama/llama/math_ops.py

```diff
@@ -70,14 +71,15 @@ def attention(self, xq, keys, values, head_dim, mask):
 
     @Profiler.profiling_decorator("softmax")
     def softmax(self, x, dim):
-        if self.use_triton:
-            return F.softmax(x, dim=-1)
+        if self.use_triton and len(x) == 2:
```

Comment:
It looks like you're trying to check the number of dimensions here, right?  `len(x)` gets the number of elements, equivalent to `x.numel()`.  I think you want `x.dim()` or `x.ndim`.

Reply:
done